### PR TITLE
chore: use different port for each mongodb-runner

### PR DESCRIPTION
**Pull request by mcasimir**
- Wraps mongodb-runner in a script that performs a proper cleanup of data and ports
- Use different ports for different tests (this wasn't actually necessary, let me know if you want me to revert it)
- Retry `npm ci` to mitigate `ECONRESET` flakes
- Upload server logs to EVG
- Move all the logs to private signed artifacts

### CODE REVIEW

**Review by lerouxb — APPROVED** [review]

File: packages/mongodb-test-server/package.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+  "name": "@mongodb-js/mongodb-test-server",
+  "productName": "@mongodb-js/mongodb-test-server Plugin",
+  "description": "Temporary package that wraps mongodb-runner to make it a bit more observable and reliable",
```
Comment (Anemy) [inline packages/mongodb-test-server/package.json:4]: Is there a ticket we can link here?

Comment (mcasimir) [reply packages/mongodb-test-server/package.json:4]: Good point!

File: packages/mongodb-test-server/bin/testserver.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+}
+
+async function sendSignalToAll(pids, sig) {
+  const runningProcesses = pids.filter(isRunning);
+
+  if (runningProcesses.length) {
+    for (const pid of runningProcesses) {
+      try {
+        console.log(`sending ${sig} to`, pid);
+        process.kill(pid, sig);
```
Comment (gribnoysup) [inline packages/mongodb-test-server/bin/testserver.js:256]: I'm curious: killing processes by using process.kill wasn't working on windows, this is why we have this code in killall-mongo.js script

https://github.com/mongodb-js/compass/blob/3da3fa3dabcdebc3b07dc8ca88b11cd8fa42d116/scripts/killall-mongo.js#LL25C23-L25C31

Is it not the issue anymore?

File: packages/data-service/package.json [hunk 3]
```diff
@@ -39,13 +39,13 @@
     "lint": "npm run eslint . && npm run prettier -- --check .",
     "check": "npm run lint && npm run depcheck",
     "check-ci": "npm run check",
-    "pretest": "testserver start dataservice --port=27021",
+    "pretest": "node ../../scripts/testserver.js start dataservice --port=27021",
```
Comment (gribnoysup) [inline packages/data-service/package.json:42]: You should be able to do `compass-scripts testserver` instead by the way (handled by `./scripts/cli.js`) which reads a bit easier

**Pull request merged**